tsrc/test/cli/test_sync.py

```diff
@@ -191,6 +192,73 @@ def test_sync_with_force(
     ).read_text() == "2", "foo should have been reset to the latest tag"
 
 
+def add_repo_unstaged(name: str, git_server: GitServer, workspace_path: Path) -> None:
+    """Add a repo to the manifest without staging this change."""
+    repo_config = {"url": git_server.get_url(name), "dest": name}
+    manifest_data = git_server.manifest.data.copy()
+    manifest_data["repos"].append(repo_config)
+    manifest_path = workspace_path / ".tsrc" / "manifest" / "manifest.yml"
+    with open(manifest_path, "w") as manifest:
+        to_write = ruamel.yaml.dump(manifest_data)
+        assert to_write
+        manifest.write(to_write)
+
+
+def test_sync_without_no_update_manifest_flag_discards_changes(
+    tsrc_cli: CLI, git_server: GitServer, workspace_path: Path
+) -> None:
+    """
+    Scenario:
+    * Create a manifest with one repo, foo
+    * Initialize a workspace from this manifest
+    * Add repo bar and do not commit or push this change
+    * Run tsrc sync
+    * The manifest is updated from the remote, and the change is discarded
+    * Only foo is present after the sync
+    """
+    git_server.add_repo("foo")
+    git_server.push_file("foo", "foo.txt", contents="foo")
+    tsrc_cli.run("init", git_server.manifest_url)
+
+    git_server.add_repo("bar", add_to_manifest=False)
+    git_server.push_file("bar", "bar.txt", contents="bar")
+    add_repo_unstaged("bar", git_server, workspace_path)
+
+    tsrc_cli.run("sync")
+
+    bar_path = workspace_path / "bar"
+    assert not bar_path.exists(), "bar should not have been synced"
+
+
+def test_sync_with_no_update_manifest_flag_leaves_changes(
+    tsrc_cli: CLI, git_server: GitServer, workspace_path: Path
+) -> None:
+    """
+    Scenario:
```

Comment:
The scenarios are a bit convoluted.  You can just check the state of the git repo in `.tsrc/manifest` using `tsrc.git.status`

No need for running `tsrc sync`.

Reply:
I don't get it. Why don't I need to run `tsrc sync` when that's exactly what I'm testing?

Reply:
Sorry -you're right.

I was thinking  you could have a simpler scenario, like this:

* Modify the local manifest
* Run `tsrc sync --no-update-manifest`
* Check that the diff is still there

But that won't really work unless the diff still produces a*valid* manifest...

Carry on :)